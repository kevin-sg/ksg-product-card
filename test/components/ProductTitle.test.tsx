import React from 'react';
import '../setupTest';
import { mount } from 'enzyme';

import ProductCard, { ProductTitle } from '../../src/components';
import { product1 } from '../data/products';

describe('ProductTitle', () => {
  test('Debe mostrar el componente correctamente con el titulo', () => {
    const wrapper = mount(<ProductTitle title="Custom Product" className="custom-class" />);
    expect(wrapper).toMatchSnapshot();
  });

  test('Debe mostrar el componente con el nombre del producto', () => {
    const wrapper = mount(<ProductCard product={product1}>{() => <ProductTitle />}</ProductCard>);
    expect(wrapper).toMatchSnapshot();
  });
});
