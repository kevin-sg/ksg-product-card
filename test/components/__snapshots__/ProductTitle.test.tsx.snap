// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductTitle Debe mostrar el componente con el nombre del producto 1`] = `
<div
  className="productCard "
>
  <span
    className="productDescription "
  >
    Coffee Mug - Sin imagen
  </span>
</div>
`;

exports[`ProductTitle Debe mostrar el componente correctamente con el titulo 1`] = `
<span
  className="productDescription custom-class"
>
  Custom Product
</span>
`;
